src/TConsole.cpp

```diff
@@ -1973,11 +1978,29 @@ int TConsole::select(const QString& text, int numOfMatch)
     P_end.setY(mUserCursor.y());
 
     if (mudlet::debugMode) {
-        TDebug(QColor(Qt::darkRed), QColor(Qt::black)) << "P_begin(" << P_begin.x() << "/" << P_begin.y() << "), P_end(" << P_end.x() << "/" << P_end.y()
-                                                       << ") selectedText = " << buffer.line(mUserCursor.y()).mid(P_begin.x(), P_end.x() - P_begin.x()) << "\n"
+        TDebug(QColor(Qt::darkRed), QColor(Qt::black)) << QStringLiteral("P_begin(%1/%2), P_end(%3/%4) selectedText:\n\"%5\"\n")
+                                                                  .arg(QString::number(P_begin.x()),
+                                                                       QString::number(P_begin.y()),
+                                                                       QString::number(P_end.x()),
+                                                                       QString::number(P_end.y()),
+                                                                       buffer.line(mUserCursor.y()).mid(P_begin.x(), P_end.x() - P_begin.x()))
                 >> 0;
     }
-    return begin;
+    return qMakePair(begin, QString());
+}
+
+QPair<int, QString> TConsole::select(const QString& name, const QString& text, const int numOfMatch)
```

Comment:
Really cool method!

Reply:
I will be moving a lot of methods out of the `mudlet` class to the `TConsole` one where there will be a method that does the actual code for the current `TConsole` and a second one which takes an addition `const QString& name` to work out which `TConsole` to call - as some do already.

This has the benefit that if/when we derive a `TMainConsole` class that inherits/is derived from the `TConsole` class then it can have *all* the methods that take the extra `QString` reference and also remove the need for all the members that are only used in the `mType == TConsole::MainConsole` instance currently from being in **every** `TConsole`.